src/pages/admin/ReportDashboard.tsx

```diff
@@ -0,0 +1,158 @@
+import { useEffect, useState } from 'react';
+import {
+  Input,
+  Select,
+  Spinner,
+  Text,
+  VStack,
+  Box,
+  Heading,
+  Collapse,
+  Button,
+  useToast,
+  HStack,
+} from '@chakra-ui/react';
+import { getAllReport } from '../../services/report';
+import { Report, ReportType } from '../../types';
+import { devPrint } from '../../components/utils/RandomUtils';
+
+const ReportDashboard = () => {
+  const [reports, setReports] = useState<Report[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [searchKeyword, setSearchKeyword] = useState('');
+  const [filterType, setFilterType] = useState<ReportType>();
+  const [filterUserId, setFilterUserId] = useState<number>();
+  const [expandedReportId, setExpandedReportId] = useState<string>();
+  const toast = useToast();
+
+  useEffect(() => {
+    setLoading(true);
+    getAllReport()
+      .then((res) => {
+        setReports(res);
+        setLoading(false);
+      })
+      .catch((error) => {
+        devPrint(error);
+        toast({
+          title: 'Error',
+          description: 'Failed to fetch reports',
+          status: 'error',
+          duration: 3000,
+          isClosable: true,
+        });
+        setLoading(false);
+      });
+  }, [toast]);
+
+  const filteredReports = reports.filter((report) => {
+    return (
+      (filterType || report.type === filterType) &&
```

Comment:
I don't think this works, if you have a filterType the 2nd condition won't ever be checked? I think it needs to be:

`!filterType || report.type === filterType`